### PR TITLE
move imports to use dagster._legacy

### PR DESCRIPTION
**Pull request by dpeng817**
Moves a bunch of imports to use dagster._legacy. I performed this using a codemod tool called libCST. The code can be found here: https://github.com/dpeng817/codemods.

### CODE REVIEW

File: examples/docs_snippets/docs_snippets/concepts/assets/asset_group.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # pylint: disable=redefined-outer-name
 # start_marker
-from dagster import AssetGroup, asset
+from dagster import asset
+from dagster._legacy import AssetGroup
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/concepts/assets/asset_group.py:4]: isn't actually exposed in docs

File: examples/docs_snippets/docs_snippets/concepts/assets/from_package_name.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from dagster import AssetGroup
+from dagster._legacy import AssetGroup
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/concepts/assets/from_package_name.py:1]: isn't actually exposed in docs

File: examples/docs_snippets/docs_snippets/concepts/dagit/graphql/client_example.py [hunk 3]
```diff
@@ -95,6 +94,7 @@ def do_something_with_exc(some_exception):  # pylint: disable=W0613
     ShutdownRepositoryLocationInfo,
     ShutdownRepositoryLocationStatus,
 )
+from dagster._legacy import PipelineRunStatus
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/concepts/dagit/graphql/client_example.py:97]: PR out to fix this https://github.com/dagster-io/dagster/pull/9055

File: examples/docs_snippets/docs_snippets/concepts/repositories_workspaces/lazy_repository_definition.py [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 import datetime
 
-from dagster import InputDefinition, RunRequest, daily_schedule, job, repository, sensor
-from dagster._legacy import pipeline, solid
+from dagster import RunRequest, daily_schedule, job, repository, sensor
+from dagster._legacy import InputDefinition, pipeline, solid
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/concepts/repositories_workspaces/lazy_repository_definition.py:6]: not referenced in actual docs

File: examples/docs_snippets/docs_snippets/legacy/data_science/download_file.py [hunk 5]
```diff
@@ -5,7 +5,8 @@
 
 from urllib.request import urlretrieve
 
-from dagster import Field, OutputDefinition, String, op
+from dagster import Field, String, op
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/legacy/data_science/download_file.py:8]: handled by https://github.com/dagster-io/dagster/pull/9055

File: examples/memoized_development/memoized_development/repo.py [hunk 6]
```diff
@@ -3,10 +3,10 @@
 from memoized_development.solids.sentence import emit_sentence
 from memoized_development.solids.tree import emit_tree
 
-from dagster import ModeDefinition, repository
```
Comment (dpeng817) [inline examples/memoized_development/memoized_development/repo.py:6]: legacy example to be removed by https://github.com/dagster-io/dagster/pull/9055

**Review by smackesey — APPROVED** [review]
lgtm

**Review by sryza — APPROVED** [review]

**Pull request merged**